src/core/state.go

```diff
@@ -34,13 +34,19 @@ type ParseTask struct {
 	ForSubinclude    bool
 }
 
-// BuildTask is the type for the build task queue
-type BuildTask = BuildLabel
+// A TaskType identifies whether a task is a build or test action.
+type TaskType uint16
 
-// TestTask is the type for the test task queue
-type TestTask struct {
+const (
+	BuildTask TaskType = 0
+	TestTask  TaskType = 1
+)
+
+// A Task is the type for the queue of build/test tasks.
+type Task struct {
 	Label BuildLabel
-	Run   int
+	Type  TaskType
+	Run   int32 // Only present for tests (the run of a build is always zero)
```

Comment:
nit: unsigned variant better describes this?

Reply:
sure thing - also lets us handle up to 4 billion runs, just in case!